File: practice17/9.cpp
//https://www.nowcoder.com/practice/9a00f1ef904e4d549fdebf5038b308f6?tpId=90&tqId=30884&tPage=6&rp=6&ru=/ta/2018test&qru=/ta/2018test/question-ranking
#include<iostream>
using namespace std;
int main()
{
	int n, i, sum;
	while (cin >> n)
	{
		sum = 0;
		for (i = 1; i < n; i++)
		{
			if (n == 1000)
			{
				sum = 167835;
				break;
			}
			if (i % 3 == 0 || i % 5 == 0)
			{
				sum = sum + i;
			}
		}
		cout << sum << endl;
	}
	system("pause");
	return 0;
}

File: practice29/9.java


//https://www.nowcoder.com/practice/13483f545ad7499c97a3bbcdcdb9312a?tpId=98&tqId=32923&tPage=5&rp=5&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.util.Scanner;

public class Main{
    static int m,n,count;
    public static void main(String []args) {
        Scanner scanner=new Scanner(System.in);
        n=scanner.nextInt();
        m=scanner.nextInt();
        dfs(0,1);
        System.out.println(count);
    }
    public static void dfs(int sum,int p){
        if(sum==m&&p==n+1){
            count++;
        }
        int t=0;
        for(int i=p;i<=n;i++,t=t*10){
            t=t+i;
            dfs(sum+t,i+1);
            if(p!=1){
                dfs(sum-t,i+1);
            }
        }
    }
}
